src/test/java/uk/gov/companieshouse/acsp/manage/users/integration/UserAcspMembershipIntegrationTest.java

```diff
@@ -720,880 +728,400 @@ void addAcspMemberSuccessWhenInviteeHasRemovedMembershipInDifferentAcsp() throws
+      TestDataManager testDataManager = TestDataManager.getInstance();
+
+      AcspMembersDao witActingOwner = testDataManager.fetchAcspMembersDaos("WIT004").getFirst();
+      AcspMembersDao witOwner = testDataManager.fetchAcspMembersDaos("WIT001").getFirst();
+      AcspMembersDao witAdmin = testDataManager.fetchAcspMembersDaos("WIT002").getFirst();
+      AcspMembersDao witStandard = testDataManager.fetchAcspMembersDaos("WIT003").getFirst();
+
+      AcspMembersDao neiActingAdmin = testDataManager.fetchAcspMembersDaos("NEI004").getFirst();
+      AcspMembersDao neiOwner = testDataManager.fetchAcspMembersDaos("NEI001").getFirst();
+      AcspMembersDao neiAdmin = testDataManager.fetchAcspMembersDaos("NEI002").getFirst();
+      AcspMembersDao neiStandard = testDataManager.fetchAcspMembersDaos("NEI003").getFirst();
+
+      AcspMembersDao xmeActingStandard = testDataManager.fetchAcspMembersDaos("XME004").getFirst();
+      AcspMembersDao xmeOwner = testDataManager.fetchAcspMembersDaos("XME001").getFirst();
+      AcspMembersDao xmeAdmin = testDataManager.fetchAcspMembersDaos("XME002").getFirst();
+      AcspMembersDao xmeStandard = testDataManager.fetchAcspMembersDaos("XME003").getFirst();
+
+      return Stream.of(
+              // Owner actions
+              Arguments.of(witActingOwner, witOwner, status().isOk()),
+              Arguments.of(witActingOwner, witAdmin, status().isOk()),
+              Arguments.of(witActingOwner, witStandard, status().isOk()),
+
+              // Admin actions
+              Arguments.of(neiActingAdmin, neiOwner, status().isBadRequest()),
+              Arguments.of(neiActingAdmin, neiAdmin, status().isOk()),
+              Arguments.of(neiActingAdmin, neiStandard, status().isOk()),
+
+              // Standard member actions
+              Arguments.of(xmeActingStandard, xmeOwner, status().isBadRequest()),
+              Arguments.of(xmeActingStandard, xmeAdmin, status().isBadRequest()),
+              Arguments.of(xmeActingStandard, xmeStandard, status().isBadRequest())
+      );
+    }
 
-    mockMvc.perform( patch( "/acsp-members/NEI003" )
-                    .header("X-Request-Id", "theId123")
-                    .header("Eric-identity", "67ZeMsvAEgkBWs7tNKacdrPvOmQ")
-                    .header("ERIC-Identity-Type", "oauth2")
-                    .header("ERIC-Authorised-Key-Roles", "*")
-                    .contentType( MediaType.APPLICATION_JSON )
-                    .content( "{\"action\":\"edit_role\",\"user_role\":\"standard\"}" ) )
-            .andExpect( status().isOk() );
+    @ParameterizedTest
+    @MethodSource("membershipRemovalScenarios")
+    void testUpdateAcspMembershipRemovalScenarios(
+        AcspMembersDao actor, AcspMembersDao target, ResultMatcher expectedStatus)
+        throws Exception {
+      // given
+      acspMembersRepository.insert(List.of(actor, target));
+      final var actingUser = new User();
+      actingUser.setUserId(actor.getUserId());
```

Comment:
Great job, I really like the refactoring you've done! 

I think it would be good if we could have just one more minor change. 

Instead of creating a new User, it would be good if we could use:

testDataManger.fetchUserDtos( actor.getUserId() ).getFirst()